internal/ethapi/api_test.go

```diff
@@ -730,36 +730,42 @@ func TestRPCMarshalBlock(t *testing.T) {
 		},
 	}
 
 	for i, tc := range testSuite {
-		resp := RPCMarshalBlock(block, tc.inclTx, tc.fullTx, params.MainnetChainConfig)
+		resp := NewRPCBlock(block, tc.inclTx, tc.fullTx, params.MainnetChainConfig)
 		out, err := json.Marshal(resp)
 		if err != nil {
 			t.Errorf("test %d: json marshal error: %v", i, err)
 			continue
 		}
-		if have := string(out); have != tc.want {
-			t.Errorf("test %d: want: %s have: %s", i, tc.want, have)
-		}
+
+		want, have := tc.want, string(out)
+		require.JSONEqf(t, want, have, "test %d: json not match, want: %s, have: %s", i, want, have)
```

Comment:
Why changing the raw string comparison into a json-equality check?

Reply:
seems the JSON serialization order is differ